drivers/ninaw10/nina_wifi_bsp.c

```diff
@@ -46,67 +46,82 @@
 #define debug_printf(...)
 #endif
 
-int nina_bsp_init(void) {
-    mp_hal_pin_output(MICROPY_HW_NINA_GPIO1);
+// Use the name CS instead of GPIO1
+#ifndef MICROPY_HW_NINA_CS
+#define MICROPY_HW_NINA_CS MICROPY_HW_NINA_GPIO1
+#endif
+
+MP_WEAK int nina_bsp_init(void) {
```

Comment:
I don't think any of these `MP_WEAK` are needed.  It seems that the only function that's overridden by mimxrt is `nina_bsp_spi_slave_select`, and the only difference I could see there is that `mp_hal_delay_ms(1)` is replaced with `mp_hal_delay_us(100)`.  Is that change necessary?  If so, maybe it should be changed to `mp_hal_delay_us(100)` for everyone?

Reply:
Testing performance with the 1ms delay shows a drop of 15-20% vs the 100µs delay. That might not be worth the change. WiFi is slow using the NINA firmware, but at least it works.